### PR TITLE
CRDCDH-678 Data Submission Review Comments

### PR DESCRIPTION
**Pull request by Alejandro-Vega**
### Overview

Added Review Comments button in Data Submission Dashboard. It will only show latest comment for Rejections.

### Change Details (Specifics)

- Added Review Comments button
- Disabled Review Comments button if not comment is available, instead of hiding button
- Only Rejection reviewComments should be displayed, Admin submit reviewComments should never be shown
- Updated spacing and styling in Summary section
- Reused Review Comments dialog from Questionnaire, with some changes and moved to a shared folder
- Updated questionnaire to use new shared component for dialog
- Updated questionnaire tests
- Added tests for Data Submission Summary section and new Review Comments dialog component 

### Related Ticket(s)

[CRDCDH-678](https://tracker.nci.nih.gov/browse/CRDCDH-678)

### CODE REVIEW

File: src/components/DataSubmissions/DataSubmissionSummary.tsx [hunk 1]
```diff
@@ -42,35 +44,63 @@ const StyledSubmissionStatus = styled(Typography)(() => ({
...
+  }
+}));
+
+const StyledHistoryButton = styled(Button)(() => ({
+  "&.MuiButton-root": {
+    minWidth: "168px",
     marginBottom: "10px",
-    padding: "10px 20px",
+    padding: "11px 20px",
     border: "1px solid #004A80",
```
Comment (amattu2) [inline src/components/DataSubmissions/DataSubmissionSummary.tsx:79]: Can you double check if #004A80 is the correct border color? The Figma mockup seems to use the same #B3B3B3 color that the Review Comments button has.

Ours:

Screenshot 2024-02-16 at 9 59 49 AM

Figma:

Screenshot 2024-02-16 at 10 01 44 AM

File: src/components/DataSubmissions/DataSubmissionSummary.tsx [hunk 2]
```diff
@@ -172,17 +217,26 @@ const DataSubmissionSummary: FC<Props> = ({ dataSubmission }) => {
...
+            </StyledReviewCommentsButton>
+            <StyledHistoryButton
+              variant="contained"
+              color="info"
+              onClick={handleOnHistoryDialogOpen}
+            >
+              Full History
+            </StyledHistoryButton>
+          </Stack>
         </Stack>
```
Comment (amattu2) [inline src/components/DataSubmissions/DataSubmissionSummary.tsx:240]: Could you confirm if our alignment is correct? Figma seems to have it slightly more centered under the status, whereas ours is offset to the right. Maybe this is just due to the length of the status text?

Ours:

Screenshot 2024-02-16 at 10 03 03 AM

Figma:

Screenshot 2024-02-16 at 10 03 13 AM

**Review by amattu2 — CHANGES_REQUESTED** [review]
Awesome job (as always). Two design related questions added, and only one code-related issue.

This warning/error is appearing in the console:

```
Warning: React does not recognize the `getColorScheme` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `getcolorscheme` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
```

Everything else LGTM.

File: src/components/DataSubmissions/DataSubmissionSummary.tsx (hunk 2 above)
Comment (Alejandro-Vega) [reply src/components/DataSubmissions/DataSubmissionSummary.tsx]: Thanks for catching these for me, fixed them.

**Review by amattu2 — APPROVED** [review]

**Pull request merged**